bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitChangeListener.java

```diff
@@ -358,35 +357,60 @@ private void createRootAccessories(Item item) {
-            }
+        final @Nullable Map<String, Object> itemConfiguration = HomekitAccessoryFactory.getItemConfiguration(item,
+                metadataRegistry);
+        if (accessoryTypes.isEmpty() || !(groups.isEmpty() || groups.stream().noneMatch(g -> g.getBaseItem() == null))
+                || !itemIsForThisBridge(item, itemConfiguration)) {
+            return;
+        }
+
+        final HomekitAccessoryType primaryAccessoryType = getPrimaryAccessoryType(item, accessoryTypes,
+                itemConfiguration);
+        logger.trace("Item {} is a HomeKit accessory of types {}. Primary type is {}", item.getName(), accessoryTypes,
+                primaryAccessoryType);
+        final HomekitOHItemProxy itemProxy = new HomekitOHItemProxy(item);
+        final HomekitTaggedItem taggedItem = new HomekitTaggedItem(new HomekitOHItemProxy(item), primaryAccessoryType,
+                itemConfiguration);
+        try {
+            final HomekitAccessory accessory = HomekitAccessoryFactory.create(taggedItem, metadataRegistry, updater,
+                    settings);
+
+            accessoryTypes.stream().filter(aType -> !primaryAccessoryType.equals(aType.getKey()))
+                    .forEach(additionalAccessoryType -> {
+                        final HomekitTaggedItem additionalTaggedItem = new HomekitTaggedItem(itemProxy,
+                                additionalAccessoryType.getKey(), itemConfiguration);
+                        try {
+                            final HomekitAccessory additionalAccessory = HomekitAccessoryFactory
+                                    .create(additionalTaggedItem, metadataRegistry, updater, settings);
+                            accessory.getServices().add(additionalAccessory.getPrimaryService());
+                        } catch (HomekitException e) {
+                            logger.warn("Cannot create additional accessory {}", additionalTaggedItem);
+                        }
+                    });
+            accessoryRegistry.addRootAccessory(taggedItem.getName(), accessory);
+        } catch (HomekitException e) {
+            logger.warn("Cannot create accessory {}", taggedItem);
+        }
+    }
+
+    private boolean itemIsForThisBridge(Item item, @Nullable Map<String, Object> configuration) {
+        // non-tagged accessories belong to the first instance
+        if (configuration == null) {
+            return (instance == 1);
+        }
+
+        final @Nullable Object value = configuration.get(HomekitTaggedItem.INSTANCE);
+        if (value == null && instance == 1) {
+            return true;
+        }
+        if (value == null) {
+            return false;
+        }
+        if (value instanceof Number) {
+            return (instance == ((Number) value).intValue());
         }
+        logger.warn("Unrecognized instance tag {} ({}) for item {}; not creating accessory.", value, value.getClass(),
+                item.getName());
+        return false;
```

Comment:
would it not be better to assign wrong tagged items to first instance? (while keeping the warning)

Reply:
I don't have strong feelings either way. I initially did it this way because of PTSD of going over the limit and _everything_ disappearing from Home on the phone, and then have to re-assign all the items to their rooms. But yeah, I think you're right it's better to have them show up _somewhere_.